### DISCUSSION TITLE
Getting strange `;` at the end of HTML output

### DISCUSSION THREAD

**Discussion by dukedhx** [vercel/next.js#11321] [Help] [answered]
Anyone has any ideas why I am getting a mysterious semicolon at the end of the page?
![image](https://user-images.githubusercontent.com/10786558/77415839-00c12480-6dfe-11ea-953f-bcaa25b53dd6.png)

Nowhere to find it in any of the code though:
```
//document.ts
export default class CustomDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const sheets = new ServerStyleSheets()
    const originalRenderPage = ctx.renderPage

    ctx.renderPage = () =>
      originalRenderPage({
        enhanceApp: App => props => sheets.collect(<App {...props} />)
      })
    const initialProps = await Document.getInitialProps(ctx)

    return {
      ...initialProps,
      styles: [
        ...React.Children.toArray(initialProps.styles),
        sheets.getStyleElement()
      ]
    }
  }

  render() {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
```

```
//index.ts
  return (
    <>
      <Head>
        <title>233</title>
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      </Head>
      <Layout maxWidth='xl' background='none' headerSettings={{menus,asyncSearch:true,showFilter:false}}>
       <div className={showContents?classes.visible:classes.invisible}>
        {pagination}
        <div
          ref={container}
          style={{
            position: 'relative',
          }}
        >
          {currentEntries.length > 0 && renderPostList()}
        </div>
        {pagination}
        </div>
      </Layout>
    </>
  )
}
```

Any ideas? Thanks!

**Comment by timneutkens** [comment]
Do you have a site url?

**Reply by BhavyaCodes** [reply]
Im getting the same issue - https://portfolio-git-add-navigation-buttons-juggernaut9.vercel.app/
source - https://github.com/Juggernaut9/portfolio/tree/add-navigation-buttons

**Reply by BhavyaCodes** [reply]
Nevermind i found it

![image](https://user-images.githubusercontent.com/58269749/112383062-a7dae380-8d12-11eb-9541-065acdec7dee.png)

**Reply by imabp** [reply]
Thank you @BhavyaCodes you saved me 4 hours of debugging xD.
I dont know how it appeared, but more suspicion about `prettier --write`

**Reply by BhavyaCodes** [reply]
> Thank you @BhavyaCodes you saved me 4 hours of debugging xD. I dont know how it appeared, but more suspicion about `prettier --write`

Yes it happens sometimes due to prettier which does it job fine but you neglect the semicolon when modifying code after that, need to be careful about that extra `;`

**Reply by BerkanERDAL** [reply]
thanks i was lookin for that

**Reply by tieming-yang** [reply]
Thank you!

**Reply by Vagabondideas** [reply]
Ditto. Anoying semicolon found at the same place. Thank you!

**Reply by Rozygeldi94** [reply]
> Nevermind i found it
> 
> ![image](https://user-images.githubusercontent.com/58269749/112383062-a7dae380-8d12-11eb-9541-065acdec7dee.png)

thanks bro

**Reply by biratdatta** [reply]
Thanks Bro, even I had this too.

**Comment by dukedhx** [comment, marked as answer]
Nevermind just sorted it myself - it was a trailing semicolon that sneaked into the SEO component from nowhere ...

**Reply by BhavyaCodes** [reply]
Hey, Im unable to find it lol. Can you help?
https://github.com/vercel/next.js/discussions/11321#discussioncomment-526842

**Reply by BhavyaCodes** [reply]
hey, i found it

**Comment by biratdatta** [comment]
![image](https://github.com/user-attachments/assets/79102dea-e5a0-4b78-9282-3c16612b3218)
Just remove the semicolon at the end of the props.
